### DISCUSSION TITLE
Action being cancelled by timeout does not run failure step

### DISCUSSION THREAD

**Discussion by retorquere** [community/community#25788] [Actions]
<p>I have an action that gets cancelled because it runs too long. I have a step with condition with <code>if: ${{ failure() }}</code> to upload the logs for diagnosis, but when an action is aborted, this step does not run. Can I give another condition that will cause this step to run?</p>

**Comment by jsoref** [comment]
<p>Try setting a lower timeout?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>There’s a separate <code>cancelled()</code> status check, see <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#cancelled" class="inline-onebox">Expressions - GitHub Docs</a>.</p>
